### PR TITLE
Made ColorPicker save it's presets while in editor.

### PR DESCRIPTION
**Pull request by Piet-G**
Made ColorPicker save it's presets while in editor.

Fixes #21169 and #5427

### CODE REVIEW

**Comment by merumelu** [comment]
ColorPicker is not editor only, it fails building for `tools=no` because `EditorSettings` is not available: https://travis-ci.org/godotengine/godot/jobs/432499722#L1986

Also I think it would be nicer if this was an actual method, something like `save_presets_to_file(path)` so you can use it for games as well.

**Comment by DualMatrix** [comment]
Made sure it only saved while in the editor.

File: scene/gui/color_picker.cpp [hunk 1]
```diff
@@ -186,9 +201,18 @@ void ColorPicker::_update_presets() {
...
 	preset->draw_texture_rect(get_icon("preset_bg", "ColorPicker"), Rect2(Point2(), preset_size), true);
 
-	for (int i = 0; i < presets.size(); i++) {
-		preset->draw_rect(Rect2(Point2(size.width * i, 0), size), presets[i]);
+#ifdef TOOLS_ENABLED
+	if (Engine::get_singleton()->is_editor_hint()) {
+		PoolColorArray arr_to_save = PoolColorArray();
+
+		for (int i = 0; i < presets.size(); i++) {
+			preset->draw_rect(Rect2(Point2(size.width * i, 0), size), presets[i]);
```
Comment (merumelu) [inline scene/gui/color_picker.cpp]: Like this added presets will only show up in the editor. This needs to be outside the ifdef.

Comment (DualMatrix) [reply scene/gui/color_picker.cpp]: Good catch, I need to pay more attention... Fixed

**Pull request merged**